backend/backend.py

```diff
@@ -29,114 +29,93 @@
 
 @app.errorhandler(404)
 def page_not_found():
-    return "<h1>404</h1><p>Page Not Found.</p>", 404
+    return '<h1>404</h1><p>Page Not Found.</p>', 404
 
 @app.route('/')
 def index():
     return redirect(url_for('login'))
 
 
+#register should contain fields:
+# username, password, email
 @app.route('/signup', methods=['POST'])
 def register():  
     if request.method == 'POST':
-        new_user = request.get_json()
-        print("\n" ,new_user, "\n")
+        req = request.get_json()
 
         # Check that all required fields were entered
-        if new_user['username'] and new_user['password']:
-            if User().get(new_user['username']): # Check if user exists
-                # flash('This Username already exists!')
-                print("HERE")
-                return jsonify({'error': 'This Username already exists!'}), 409
-            pw_salt = bcrypt.gensalt()
-            encrypted_password = bcrypt.hashpw(new_user['password'].encode('utf8'), pw_salt)
-            user_to_add = {
-                'username': new_user['username'],
-                'password': encrypted_password,
-                'email': new_user['email'],
-                'posts': list(),
-                'likes': list(),
-                'address': dict(),
-            }
-            user_to_add = User(user_to_add)
-            user_to_add.save()
-            return jsonify(new_user), 201
-
-
-# this method will need to be called by the method the queries the database
-def validate_password(pw, db_pw):
-    return bcrypt.checkpw(pw.encode('utf8'), db_pw)
+        if not (req['username'] and req['password'] and req['email']):
```

Comment:
I changed some of this in my commit as well but I like your approach better. We'll just have to deal with a couple merge conflicts